ui/next/app/containers/banner/helpusBanner.tsx

```diff
@@ -0,0 +1,86 @@
+/// <reference path="../../../typings/index.d.ts" />
+import * as React from "react";
+import { connect } from "react-redux";
+import { Link, RouterOnContext } from "react-router";
+import _ = require("lodash");
+
+import Banner from "./banner";
+import { KEY_HELPUS, OptInAttributes, KeyValue, loadUIData, saveUIData } from "../../redux/uiData";
+
+class HelpusBannerProps {
+  attributes: OptInAttributes;
+  attributesLoaded: boolean;
+  loadUIData: (...keys: string[]) => void;
+  saveUIData: (...values: KeyValue[]) => void;
+}
+
+class HelpusBannerState {
+  dismissed: boolean = false;
+}
+
+interface HelpusBannerContext {
+  router: RouterOnContext;
+}
+
+class HelpusBanner extends React.Component<HelpusBannerProps, HelpusBannerState> {
+  static contextTypes: React.ValidationMap<any> = {
+    router: React.PropTypes.object,
+  };
+
+  state = new HelpusBannerState();
+  context: HelpusBannerContext;
+
+  componentWillMount() {
+    if (this.context && this.context.router && this.context.router.isActive("/help-us/reporting")) {
```

Comment:
Conversely, why should the helpus banner know about the routing system?  To me, this looks way more like spaghetti to me; it's doing conditional logic on a "global" setting which is difficult to access.